libs/core-ui/src/lib/util/JointDataset.ts

```diff
@@ -645,6 +645,34 @@ export class JointDataset {
     });
   }
 
+  public getJointDatasetFeatureName(userFeatureName: string | undefined): string | undefined {
+    // Return the joint dataset feature name for the given user feature name.
+    if (userFeatureName === undefined) {
+      return undefined;
+    }
+    let jointDatasetFeatureName = undefined;
+    let userDatasetFeatureName = undefined;
+    for (jointDatasetFeatureName in this.metaDict) {
+      if (
+        this.metaDict[jointDatasetFeatureName]
+          .abbridgedLabel === userFeatureName
+      ) {
+        userDatasetFeatureName =
+          this.metaDict[jointDatasetFeatureName]
+            .abbridgedLabel;
+        break;
+      }
+    }
+
+    if (
+      userDatasetFeatureName !== undefined &&
+      jointDatasetFeatureName !== undefined
+    ) {
+      return jointDatasetFeatureName;
+    }
+    return undefined;
```

Comment:
Perhaps it might be better to just throw in that case, since this is not expected when calling this utility method?

Reply:
I don't want throw error as this will leave the dashboard in an unusable state. The undefined is to be handled by the caller gracefully.